doc/source/train/huggingface-accelerate.rst

```diff
@@ -3,31 +3,201 @@
+        and pass it to the Accelerator.
+
+        .. code-block:: python
+
+            from torch.distributed.fsdp.fully_sharded_data_parallel import FullOptimStateDictConfig, FullStateDictConfig
+            from accelerate import Accelerator, FullyShardedDataParallelPlugin
+
+            def train_func(config):
+                fsdp_plugin = FullyShardedDataParallelPlugin(
+                    state_dict_config=FullStateDictConfig(
+                        offload_to_cpu=False, 
+                        rank0_only=False
+                    ),
+                    optim_state_dict_config=FullOptimStateDictConfig(
+                        offload_to_cpu=False, 
+                        rank0_only=False
+                    )
+                )
+
+                # Initialize accelerator
+                accelerator = Accelerator(
+                    ...,
+                    fsdp_plugin=fsdp_plugin,
+                )
+
+                # Start training
+                ...
+
+            from ray.train.torch import TorchTrainer
+            from ray.train import ScalingConfig
+
+            trainer = TorchTrainer(
+                train_func,
+                scaling_config=ScalingConfig(...),
+                ...
+            )
+            trainer.fit()
+
+Next, check these end-to-end examples below for more details:
+
+.. dropdown:: End-to-end Code Example
+
+    .. literalinclude:: /../../python/ray/train/examples/accelerate/accelerate_torch_trainer.py
         :language: python
+
+.. seealso::
+
+    If you're looking for more advanced use cases, check out this Llama-2 fine-tuning example: 
+    
+    - `Fine-tuning Llama-2 series models with Deepspeed, Accelerate, and Ray Train. <https://github.com/ray-project/ray/tree/master/doc/source/templates/04_finetuning_llms_with_deepspeed>`_
+
+You may also find these user guides helpful:
+
+- :ref:`Configuring Scale and GPUs <train_scaling_config>`
+- :ref:`Configuration and Persistent Storage <train-run-config>`
+- :ref:`Saving and Loading Checkpoints <train-checkpointing>`
+- :ref:`How to use Ray Data with Ray Train <data-ingest-torch>`
+
+
+`AccelerateTrainer` Migration Guide 
+-----------------------------------
+
+Before Ray 2.7, Ray Train's :class:`AccelerateTrainer <ray.train.huggingface.AccelerateTrainer>` API was the 
+recommended way to run Accelerate code. As a subclass of :class:`TorchTrainer <ray.train.torch.TorchTrainer>`,  
+AccelerateTrainer helps users apply the configuration file generated with ``accelerate config`` to all workers. 
+Aside from that, the functionality of ``AccelerateTrainer`` is identical to ``TorchTrainer``.
+
+However, users can easily configure Accelerate within their training function, so there is no need to maintain a separate Ray Trainer API.
+To provide a more simplified and flexible interface, we will deprecate ``AccelerateTrainer`` in Ray 2.8, and recommend running your 
+Accelerate code with ``TorchTrainer``. 
```

Comment:
```suggestion
However, this caused confusion around whether this was the *only* way to run Accelerate code. 
Because the full Accelerate functionality can be expressed with the ``Accelerator`` and ``TorchTrainer`` combination, the ``AccelerateTrainer`` will be deprecated in Ray 2.8, 
and it is recommend to run your  Accelerate code directly with ``TorchTrainer``. 
```